[Feature] Enable GenAI Description Generation via API for Non-Enabled Cameras

fl42 opened the pull request:
## Proposed change

This PR introduces a new optional `force` parameter (default: `false`) when triggering description generation via the API. When force is set to `true`, the system will generate a description even if GenAI is not enabled for the associated camera.

**Key Benefits**
- Delayed processing: Generate descriptions at a later time instead of in real time — useful when real-time event processing exceeds hardware capacity.
- Selective usage: Generate descriptions only for specific events based on custom logic (e.g., object size, confidence level) to save resources.

**Note**
The [UI](https://github.com/blakeblackshear/frigate/blob/ed43df9c13e7d0d67c11080101b4da1439c95cf3/web/src/components/overlay/detail/SearchDetailDialog.tsx#L937) remains unchanged — the "Regenerate Description" button is still hidden for cameras without GenAI enabled to avoid confusing users.

This feature is intended for advanced users who require fine-grained control and typically interact with the system via the API.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code
- [X] Documentation Update

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:

## Checklist

<!--
  Put an `x` in the boxes that apply.
-->

- [X] The code change is tested and works locally.
- [X] Local tests pass. **Your PR cannot be merged unless tests pass**
- [X] There is no commented out code in this PR.
- [ ] UI changes including text have used i18n keys and have been added to the `en` locale.
- [X] The code has been formatted using Ruff (`ruff format frigate`)

hawkeye217 commented:
Thanks for the PR.

First, I don't think another config option is the direction we want to take here, and it doesn't even seem like you are using the new config option in your code changes. I'll leave some separate comments on the actual lines themselves.

Second, this will need to be targeted at the 0.17 branch as we are currently only committing bugfixes to dev.

This seems like a very unique use case for GenAI, I'm curious if the other maintainers have thoughts.

hawkeye217 commented on frigate/embeddings/maintainer.py:
```diff
@@ -640,15 +640,16 @@ def _read_and_crop_snapshot(self, event: Event, camera_config) -> bytes | None:
             return
 
-        camera_config = self.config.cameras[event.camera]
-        if not camera_config.genai.enabled or self.genai_client is None:
```
Rather than another config option, I'd prefer to see the `force` param just passed through the publisher and then just checked here. For example:

`def handle_regenerate_description(self, event_id: str, source: str, force: bool = False) -> None:`

fl42 replied:
Fixed

fl42 committed:
feat: enable using GenAI for cameras with GenAI disabled from the API

fl42 commented:
Thanks for the feedback!

I've updated the PR to target the 0.17 branch and incorporated your suggestion.

I agree that this feature is primarily useful for advanced users. In my case, I'm working with limited hardware but a high volume of detections. It would be very helpful to manually control the GenAI analysis speed via the API—and ideally, to apply it selectively (e.g., skipping very small detections that are unlikely to yield useful results).

I'm open to other suggestions or alternative implementation approaches you might have.

hawkeye217 commented:
This looks good. Thanks for your contribution!
